packages/@react-stately/checkbox/src/useCheckboxGroupState.ts

```diff
@@ -34,23 +39,23 @@ export interface CheckboxGroupState {
   toggleValue(value: string): void
 }
 
-let instance = Math.round(Math.random() * 10000000000);
-let i = 0;
-
 /**
  * Provides state management for a checkbox group component. Provides a name for the group,
  * and manages selection and focus state.
  */
 export function useCheckboxGroupState(props: CheckboxGroupProps = {}): CheckboxGroupState {
-  let name = useMemo(() => props.name || `checkbox-group-${instance}-${++i}`, [props.name]);
-  let [selectedValue, setValue] = useControlledState(props.value, props.defaultValue || [], props.onChange);
+  let [selectedValues, setValue] = useControlledState(props.value, props.defaultValue || [], props.onChange);
 
   const state: CheckboxGroupState = {
-    name,
-    value: selectedValue,
+    value: selectedValues,
     setValue,
```

Comment:
if `addValue`, `removeValue` and `toggleValue` check `props.isReadOnly || props.isDisabled` before actually changing the value, shouldn't the same check apply here?